Source/inv.cpp

```diff
@@ -1636,48 +1628,44 @@ void InvGetItem(int pnum, int ii)
 		NewCursor(player.HoldItem._iCurs + CURSOR_FIRSTITEM);
 }
 
-void AutoGetItem(int pnum, Item *item, int ii)
+void AutoGetItem(int pnum, Item *itemPointer, int ii)
 {
-	bool done;
-	bool autoEquipped = false;
-
-	if (pcurs != CURSOR_HAND) {
-		return;
-	}
+	Item &item = *itemPointer;

```

Comment:
Why not take the item by reference as the parameter? There only appears to be 3 usage and it can never be null.

Reply:
To make `CleanupItem` behave like an `erase` function it should operator on an iterator, which would mean that this function would be best to take an iterator as well. So for now I was leaving the function taking a pointer until I get around to that refactoring.

Reply:
Hmm, actually that wasn't the reason exactly. CleanupItem and DeleteItem are related but CleanupItem does a bit extra to handle cornerstone items. Probably gonna end up combining the functions since they're both used for picking up an item from the ground so really should be the same thing.